### ISSUE TITLE
OSD generated as a video with transparent background

### ISSUE THREAD

**Issue by Pairan** [avsaase/walksnail-osd-tool#7]
### Is there an existing feature request for this?

- [X] I have searched the existing open and closed issues

### Feature description

Please add an option where the OSD is generated as a video with transparent background ONLY

makes it very easy to overlay on other sources like 4K footage and such

**Comment by MrD-RC** [comment]
+1 for this. I believe it would have to be done as a series as exported .PNG files with a transparent background. This is how it works with the other tool. Which makes importing the OSD in to programs like DaVinci Resolve really easy.

**Comment by MrD-RC** [comment]
I never find keying out the video to give great results. Maybe they option of of a video with a solid colour background for keying, or the png sequence.

**Comment by kirek007** [comment]
There are video formats that supports transparent background without
keying. I might be able to provide a ffmpeg example for it.. But I'm short
on time these days.
My tool was using png because it was super easy.. But it was quite big
amount of files :(

czw., 27 kwi 2023, 10:02 użytkownik Darren Lines ***@***.***>
napisał:

**Comment by Pairan** [comment]
That of course would be great @kirek007 ... and surely provide the best result

**Comment by MrD-RC** [comment]
Any news on this? This is the main thing stopping using this tool.

**Comment by Pairan** [comment]
> Any news on this? This is the main thing stopping using this tool.

As I didn't catch you on Telegram and the others might be interested ind this too: Its already done! Download the latest "nightly build" and get going! Works brilliant

**Comment by MrD-RC** [comment]
IMHO, adding a key colour is not the correct solution. Especially in a full colour capable OSD. Keying is never perfect and has artefacts. It should either output a png sequence or a video with a genuine transparent background.

**Comment by ledgendarydiy** [comment]
I don't understand how I could render without using the video file? Is there a button or check box I am missing? Or is there a certain procedure to achieve this?

**Comment by Pairan** [comment]
Yes ... you can click on the "chroma" in the bottom and then it renders the OSD against a specific color ;)
